client/src/components/Form/Reservation/index.js

```diff
@@ -0,0 +1,283 @@
+
+class BookingForm extends React.Component {
+  state = {
+    repeat: 'once',
+    confirmLoading: false,
+  };
+
+  formRef = React.createRef();
+
+  bookRoom = ({
+    repeat,
+    date,
+    daterange,
+    time,
+    remind: remindMe = true,
+    room,
+    ...rest
+  }) => {
+    const roomId = this.findRoomIdByName(room);
+    const timeArr = this.makeBookingArr(repeat, date, daterange, time);
+    this.setState({ confirmLoading: true });
+    const body = { roomId, time: timeArr, remindMe, ...rest };
+    return fetch('/api/v1/booking', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(body),
+    })
+      .then((res) => {
+        if (!res.ok) {
+          res.json().then(({ message: msg }) => message.error(msg));
+          throw res.statusText;
+        }
+        return res.json();
+      })
+      .then(() => {
+        this.setState({ confirmLoading: false });
+      })
+      .then(() => message.success('Room booked successfully', 3))
+      .catch(() => {
+        this.setState({ confirmLoading: false });
+      });
+  };
+
+  findRoomIdByName = (name) => {
+    const { rooms } = this.props;
+    return rooms.find((room) => room.name === name).id;
+  };
+
+  findRoomNameById = (id) => {
+    const { rooms } = this.props;
+    return rooms.find((room) => room.id === Number(id)).name;
+  };
+
+  repeatOnChange = (e) => {
+    this.setState({ repeat: e.target.value });
+  };
+
+  makeBookingArr = (
+    repeat,
+    date,
+    [startDate, endDate] = [],
+    [startTime, endTime]
+  ) => {
+    const arr = [];
+    if (repeat === 'weekly') {
+      for (let i = startDate; i <= endDate; i = i.add(1, 'week')) {
+        arr.push({
+          startTime: moment(
+            `${i.format('YYYY-MM-DD')} ${startTime.format('LTS')}`
+          ).toISOString(true),
+          endTime: moment(
+            `${i.format('YYYY-MM-DD')} ${endTime.format('LTS')}`
+          ).toISOString(true),
+        });
+      }
+    } else if (repeat === 'daily') {
+      for (let i = startDate; i <= endDate; i = i.add(1, 'day')) {
+        if (i.format('dddd') !== 'Friday' && i.format('dddd') !== 'Saturday') {
+          arr.push({
+            startTime: moment(
+              `${i.format('YYYY-MM-DD')} ${startTime.format('LTS')}`
+            ).toISOString(true),
+            endTime: moment(
+              `${i.format('YYYY-MM-DD')} ${endTime.format('LTS')}`
+            ).toISOString(true),
+          });
+        }
+      }
+    } else if (repeat === 'once') {
+      arr.push({
+        startTime: moment(
+          `${date.format('YYYY-MM-DD')} ${startTime.format('LTS')}`
+        ).toISOString(true),
+        endTime: moment(
+          `${date.format('YYYY-MM-DD')} ${endTime.format('LTS')}`
+        ).toISOString(true),
+      });
+    }
+    return arr;
+  };
+
+  render() {
+    const disabledDate = (current) =>
+      current && current < moment().endOf('day');
```

Comment:
We talked about this, you can disable bookings that are for example 5 mins from now